### PR TITLE
feat: introduce CommitSigner interface

### LINKED ISSUE #337: Add GPG Signature option
The GitHub commit API allows signing a commit, but the signature must be calculated externally.

https://docs.github.com/en/rest/reference/git#create-a-commit

### PR DESCRIPTION
**Pull request by chingor13**
Towards #337

This introduces an interface to allow providing a commit signer, but does not include any implementation.

### CODE REVIEW

File: src/github/create-commit.ts [hunk 1]
```diff
@@ -44,6 +60,9 @@ export async function createCommit(
       message,
       tree: treeSha,
       parents: [refHead],
+      signature,
```
Comment (bcoe) [inline src/github/create-commit.ts:63]: If you can find the REST documentation in the GitHub API, adding it here as a comment would be useful.

I'm interested to know what options we have available to us for a signing algorithm (I'm hoping we might be able to use something other than GPG).

File: test/commit-and-push.ts [hunk 2]
```diff
@@ -37,6 +44,16 @@ type CreateCommitResponse = GetResponseTypeFromEndpointMethod<
   typeof octokit.git.createCommit
 >;
 
+class FakeCommitSigner implements CommitSigner {
+  signature: string;
+  constructor(signature: string) {
+    this.signature = signature;
+  }
+  async generateSignature(_commit: CommitData): Promise<string> {
+    return this.signature;
```
Comment (bcoe) [inline test/commit-and-push.ts:53]: I would add an ignore line, to avoid this warning.

File: src/github/create-commit.ts (hunk 1 above)
Comment (chingor13) [reply src/github/create-commit.ts:64]: Added a link to the create commit API docs. It looks like only PGP is supported for this API

File: test/commit-and-push.ts (hunk 2 above)
Comment (chingor13) [reply test/commit-and-push.ts:54]: Done.

**Pull request merged**